### PR TITLE
feat(eip4844): include `TxEip4844` in the primitive `Transaction` enum 

### PR DESCRIPTION
**Pull request by tcoratger**
Should solve #4073.

### CODE REVIEW

File: crates/primitives/src/transaction/mod.rs [hunk 1]
```diff
@@ -531,6 +594,31 @@ impl Transaction {
...
+                nonce.encode(out);
+                gas_limit.encode(out);
+                max_fee_per_gas.encode(out);
+                max_priority_fee_per_gas.encode(out);
+                to.encode(out);
+                value.encode(out);
+                access_list.encode(out);
+                blob_versioned_hashes.encode(out);
+                max_fee_per_blob_gas.encode(out);
+                input.0.encode(out);
```
Comment (mattsse) [inline crates/primitives/src/transaction/mod.rs:620]: the eip defines the order as:

[chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, to, value, data, access_list, max_fee_per_blob_gas, blob_versioned_hashes, y_parity, r, s]

https://eips.ethereum.org/EIPS/eip-4844

similar to eip1559 tx type, see above

**Review by mattsse — CHANGES_REQUESTED** [review]
this is awesome,

ptal @Rjected

**Review by Rjected — COMMENTED** [review]
this looks good aside from the note about encoding

File: crates/primitives/src/transaction/mod.rs (hunk 1 above)
Comment (tcoratger) [reply crates/primitives/src/transaction/mod.rs]: @mattsse Sure you are right, I've just fixed that.

**Comment by tcoratger** [comment]
@Rjected Should be fixed, tell me if there are other stuffs or if ready to merge :)

File: crates/primitives/src/transaction/mod.rs [hunk 2]
```diff
@@ -1012,6 +1100,19 @@ impl TransactionSigned {
...
+                nonce: Decodable::decode(data)?,
+                max_priority_fee_per_gas: Decodable::decode(data)?,
+                max_fee_per_gas: Decodable::decode(data)?,
+                gas_limit: Decodable::decode(data)?,
+                to: Decodable::decode(data)?,
+                value: Decodable::decode(data)?,
+                input: Bytes(Decodable::decode(data)?),
+                access_list: Decodable::decode(data)?,
+                max_fee_per_blob_gas: Decodable::decode(data)?,
+                blob_versioned_hashes: Decodable::decode(data)?,
```
Comment (mattsse) [inline crates/primitives/src/transaction/mod.rs:1114]: this is correct

File: crates/primitives/src/transaction/mod.rs [hunk 3]
```diff
@@ -531,6 +594,31 @@ impl Transaction {
...
+                nonce.encode(out);
+                max_priority_fee_per_gas.encode(out);
+                max_fee_per_gas.encode(out);
+                gas_limit.encode(out);
+                to.encode(out);
+                value.encode(out);
+                input.0.encode(out);
+                access_list.encode(out);
+                max_fee_per_blob_gas.encode(out);
+                blob_versioned_hashes.encode(out);
```
Comment (mattsse) [inline crates/primitives/src/transaction/mod.rs:620]: also correct

**Review by mattsse — APPROVED** [review]
lgtm, pending @Rjected

**Review by Rjected — APPROVED** [review]
nice, LGTM!

**Pull request merged**